plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/debug/ui/launchConfigurations/EnvironmentTab.java

```diff
@@ -0,0 +1,156 @@
+/******************************************************************************* 
+ * Copyright (c) 2021 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.eclipse.reddeer.eclipse.debug.ui.launchConfigurations;
+
+import java.util.List;
+
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.swt.api.TableItem;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+import org.eclipse.reddeer.swt.impl.button.OkButton;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.button.RadioButton;
+import org.eclipse.reddeer.swt.impl.table.DefaultTable;
+import org.eclipse.reddeer.swt.impl.table.DefaultTableItem;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+
+/**
+ * Represents Environment tab in Launch Configuration
+ * 
+ * @author Oleksii Korniienko olkornii@redhat.com
+ *
+ */
+public class EnvironmentTab extends LaunchConfigurationTab {
+
+	private static final String ADD_SHELL_TITLE = "New Environment Variable";
+	private static final String SELECT_SHELL_TITLE = "Select Environment Variables";
+	private static final String EDIT_SHELL_TITLE = "Edit Environment Variable";
+
+	public EnvironmentTab() {
+		super("Environment");
+	}
+
+	/**
+	 * Return the list of all environment variables
+	 *
+	 * @return list of items
+	 */
+	public List<TableItem> getVariables() {
+		return new DefaultTable().getItems();
+	}
+
+	/**
+	 * Add new environment variable
+	 *
+	 * @param Name  variable name
+	 * @param Value variable value
+	 */
+	public void add(String name, String nalue) {
+		new PushButton("Add...").click();
+		new WaitUntil(new ShellIsAvailable(ADD_SHELL_TITLE));
+		new LabeledText("Name:").setText(name);
+		new LabeledText("Value:").setText(nalue);
+		new OkButton().click();
+	}
+
+	/**
+	 * Select environment variable by name
+	 *
+	 * @param selectVariable variable name for select
+	 */
+	public void select(String selectVariable) {
+		new PushButton("Select...").click();
+		new WaitUntil(new ShellIsAvailable(SELECT_SHELL_TITLE));
+		new DefaultTableItem(selectVariable).setChecked(true);
+		new OkButton().click();
+	}
+
+	/**
+	 * Select environment variable by id
+	 *
+	 * @param id variable index
+	 */
+	public void select(int id) {
```

Comment:
Can you please also add methods for selecting variables in EnvironmentTab? you have method getAllVariables, but how do I select one variable by name in this tab? Can you please also change name of current methods select(...) to selectEnvironmentVariable?

Reply:
done